### PR TITLE
[a11y] `AnchorNav` on small or zoomed in screens inaccessible elements fix

### PR DESCRIPTION
**Pull request by stamat**
## Summary

Hello there, again! 👋 

When there is a lot of items and the user is on a small 320x480 screen or zooms in the page, few last items can become inaccessible.

Adding `max-height: calc(100vh - var(--base-size-12) * 2); overflow-y: auto;` to `.AnchorNav-inner-container` fixes this issue. This adds the maximum height of the container to match the 100% of the viewport and adds an `overflow-y: auto` so the user can scroll and see all of the items.

I would love if we could 🚢  this ASAP. 🙏 

- Closes: https://github.com/github/accessibility-audits/issues/7852
- Closes: https://github.com/github/accessibility-audits/issues/8323

## List of notable changes:

- **updated** `.AnchorNav-inner-container` by adding `max-height: calc(100vh - var(--base-size-12) * 2); overflow-y: auto;`
- **removed** The custom `window.scrolTo()` was removed in favor of native anchoring and smooth scrolling. The issue was that the `event.preventDefault()` was preventing that the focus is shifted to a targeted element, and there is no substitute hack for this. We don't need the custom calculations because the browser already does them. The only downside of this is that the `#` hash anchor will appear in the browser... which is also the desired native behavior.

## Steps to test:

1. Open `/story/components-anchornav--default`
2. Reduce the screen height
3. <kbd>CMD</kbd> + <kbd>+</kbd> press several times untill the opened anchor nav is not fully visible

## Supporting resources (related issues, external links, etc):

- https://github.com/github/accessibility-audits/issues/7852
- https://github.com/github/accessibility-audits/issues/8323

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

**Review by joshfarrant — APPROVED** [review]

File: packages/react/src/AnchorNav/AnchorNav.stories.tsx [hunk 1]
```diff
@@ -119,6 +119,7 @@ export const Playground = ({data, ...args}: {data: MockData; offset: number; sec
               <Text as="p">
                 AnchorNav is a component that allows users to navigate to different sections of a page.
               </Text>
+              <Button>Learn More</Button>
```
Comment (rezrah) [inline packages/react/src/AnchorNav/AnchorNav.stories.tsx:122]: @stamat Do you need the button here? Can we remove if not, or align to start of container?

Comment (stamat) [reply packages/react/src/AnchorNav/AnchorNav.stories.tsx:122]: @rezrah I needed it to test the focus and see if it works correctly. I think it should be there for that sole purpose. But I'll align it to the start of the container! ❤️

**Comment by stamat** [comment]
@rezrah all done!

**Review by rezrah — APPROVED** [review]
Thanks @stamat ✨

**Pull request merged**